src/main/kotlin/no/nav/helsearbeidsgiver/kafka/KafkaConsumer.kt

```diff
@@ -18,12 +18,24 @@ suspend fun startKafkaConsumer(
     consumer: KafkaConsumer<String, String>,
     meldingTolker: MeldingTolker,
     enabled: () -> Boolean = { true },
+    isLeader: () -> Boolean = { false },
 ) {
     val logger = LoggerFactory.getLogger(KafkaConsumer::class.java)
+    if (isLeader()) {
+        logger.info("Pod er leder - konsumerer ikke $topic")
+        return
+    }
     consumer.subscribe(listOf(topic))
     consumer.asFlow({ consumer.toggleConsumer(enabled, topic) }).collect { record ->
         try {
+            if (isLeader()) {
```

Comment:
Sikkert greit, men hvor mange kall klarer sidecaren for leaderelection å håndtere?

Reply:
Ja, det er jo litt råflott å sjekke "hele tiden", men regner med at det går fint. 😅 
Kan jo innføre en liten cache her på sikt, eller polle med jevne mellomrom, men tenker å se hvordan det går først.